homeassistant/components/device_tracker/meraki.py

```diff
@@ -0,0 +1,137 @@
+"""
+Support for the Meraki CMX location service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.meraki/
+
+"""
+import asyncio
+import logging
+import json
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (HTTP_BAD_REQUEST, HTTP_UNPROCESSABLE_ENTITY)
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, SOURCE_TYPE_ROUTER, CONF_TRACK_NEW,
+    YAML_DEVICES, load_config, DEFAULT_TRACK_NEW)
+
+CONF_VALIDATOR = 'validator'
+CONF_SECRET = 'secret'
+DEPENDENCIES = ['http']
+URL = '/api/meraki'
+VERSION = '2.0'
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_VALIDATOR): cv.string,
+    vol.Required(CONF_SECRET): cv.string
+})
+
+
+@asyncio.coroutine
+def async_setup_scanner(hass, config, async_see, discovery_info=None):
+    """Set up an endpoint for the Meraki tracker."""
+    track_new = config.get(CONF_TRACK_NEW, DEFAULT_TRACK_NEW)
```

Comment:
You don't need to handle this at all. It's done in the device tracker base component.

Reply:
I checked https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/device_tracker/__init__.py#L273 and as I see it doesn't handle.

Reply:
It handles it here for existing devices:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/device_tracker/__init__.py#L268

And here for new devices:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/device_tracker/__init__.py#L286

Reply:
You are right. I used this option to check if device should be added to known. Since meraki floods known devices I will use other option